### PR TITLE
Fabric Go SDK Release for October 2023: v4.9 to v4.10

### PR DESCRIPTION
**Pull request by thogarty**
* Matches Public Spec 4.10 from Fabric API on SwaggerHub https://app.swaggerhub.com/apis/equinix-api/fabric/4.10
* Reverts exclusion of go.mod and go.sum from git diff verification

October Release Notes: https://uatdeveloper.equinix.com/docs?page=/dev-docs/fabric/overview

<h2 class="anchor anchorWithStickyNavbar_LWe7" id="release-overview" style="box-sizing: border-box; color: var(--ifm-heading-color); font-family: &quot;Noto Sans&quot;, Arial; font-weight: var(--ifm-heading-font-weight); line-height: var(--ifm-heading-line-height); margin-top: 0px; margin-right: ; margin-bottom: calc(var(--ifm-heading-vertical-rhythm-bottom)*var(--ifm-leading)); margin-left: ; font-size: 1.75rem; scroll-margin-top: calc(var(--ifm-navbar-height) + .5rem); --ifm-h2-font-size: 2rem; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Release Overview<a class="hash-link" href="https://uatdeveloper.equinix.com/dev-docs/fabric/release-notes/october-2023#release-overview" title="Direct link to heading" style="box-sizing: border-box; transition: opacity var(--ifm-transition-fast); color: var(--ifm-link-color); text-decoration: var(--ifm-link-decoration); opacity: 1; padding-left: 0.5rem; user-select: none;">​</a></h2><p style="box-sizing: border-box; margin-top: ; margin-right: ; margin-bottom: var(--ifm-leading); margin-left: ; color: rgb(33, 37, 41); font-family: &quot;Noto Sans&quot;, Arial; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong style="box-sizing: border-box; font-weight: var(--ifm-font-weight-bold);">Release date</strong>: October 14th, 2023<br style="box-sizing: border-box;"></p><p style="box-sizing: border-box; margin-top: ; margin-right: ; margin-bottom: var(--ifm-leading); margin-left: ; color: rgb(33, 37, 41); font-family: &quot;Noto Sans&quot;, Arial; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Equinix Fabric API specification on SwaggerHub:<span> </span><a href="https://app.swaggerhub.com/apis/equinix-api/fabric/4.9" target="_blank" rel="noopener noreferrer" style="box-sizing: border-box; transition: color var(--ifm-transition-fast) var(--ifm-transition-timing-default); color: var(--ifm-link-color); text-decoration: var(--ifm-link-decoration);"><strong style="box-sizing: border-box; font-weight: var(--ifm-font-weight-bold);">app.swaggerhub.com</strong></a><br style="box-sizing: border-box;">Equinix Fabric v4 API reference guide:<span> </span><a href="https://uatdeveloper.equinix.com/dev-docs/fabric/api-reference/fabric-v4-apis" style="box-sizing: border-box; transition: color var(--ifm-transition-fast) var(--ifm-transition-timing-default); color: var(--ifm-link-color); text-decoration: var(--ifm-link-decoration);"><strong style="box-sizing: border-box; font-weight: var(--ifm-font-weight-bold);">Fabric APIs (v4)</strong></a></p>

Feature | Summary of Release
-- | --
Routing Protocol APIs | Added new API to [Validate Subnet](https://uatdeveloper.equinix.com/dev-docs/fabric/api-reference/fabric-v4-apis#validate-subnet) 
Port APIs | Added new API to [Search Ports](https://uatdeveloper.equinix.com/dev-docs/fabric/api-reference/fabric-v4-apis#search-ports) 
Cloud Router APIs | Added "distinctIpv4PrefixesCount" and "distinctIpv6PrefixesCount" fields to [Get Specified Cloud Router](https://uatdeveloper.equinix.com/dev-docs/fabric/api-reference/fabric-v4-apis#get-specified-cloud-router)

Release Overview
Release date: October 14th, 2023

Additional Changes Not Present in Release Notes:
* Gateway Methods renamed to CloudRouter. Operation IDs in OAS were renamed causing the method rename.

Changes in SDK:
##  Enhancements
- `Account` model added
- `AdvanceConfiguration` model added
- `Connection` model updated
- `ConnectionLink` model added
- `ConnectionPostRequest` model updated
- `CloudRouterActionState` model updated
- `CloudRouterActionTypeTest` model updated
- `FabricConnectionUuid` model added
- `FabricCloudRouterCode` model updated
- `FabricCloudRouterPackages` model updated
- `FabrciCloudrouterPrice` model updated
- `GeoScopeType` model added
- `Ipv4` model added
- `Md5` model added
- `Metro` model updated
- `PackageRequest` model added
- `PortsApi` `searchPorts` method added 
- `PortSearchFieldName` model updated
- `PortV4SearchRequest` model added
- `PortsApiTest` model updated
- `PrecisiontimeApi` model added
- `PrecisionTimeChangeOperation` model added
- `PrecisionTimeServiceConnectionResponses` model added
- `PrecisionTimeServiceCreateResponse` model added
- `PrecisionTimeServicePackageResponse` model added
- `PrecisionTimeServiceRequest` model added
- `PtpAdvanceConfiguration` model added
- `Price` model updated
- `RoutesApi` model removed
- `RoutingProtocolApi` model updated
- `ValidateConnectionRequestFilter` model deleted
- `ValidateSubnetRequest` model added
- `ValidateSubnetRequestFilter` model updated
- `ValidateSubnetRequestFilterAnd` model updated
- `ValidateSubnetResponse` model added

## Breaking Changes
- `CloudRouterAPI` `*Gateway*` methods renamed to `*CloudRouter*`
- `ValidateConnectionRequestFilter` renamed to `ValidateSubnetRequestFilter`
- `SearchServiceProfiles` method signature updated to include a third argument of type *ServiceProfilesApiSearchServiceProfilesOpts

### CODE REVIEW

File: fabric/v4/model_route_table_entry_filters.go
Comment (ctreatma) [inline fabric/v4/model_route_table_entry_filters.go:6]: Out of scope for this PR, but it might be nice to leave this version out of the header comment in the generated code files (or to remove the header comment entirely, since it seems at a glance to be the same across all files).  That would eliminate some unnecessary changes in future sync PRs and make it easier to focus on the actual code changes.

File: fabric/v4/model_cloud_router_action_type.go
Comment (ctreatma) [inline fabric/v4/model_cloud_router_action_type.go:18]: My understanding is that this is a new/beta feature and breaking changes are expected at any time, but in general, in case something like this change is planned for a production feature:  enum changes like this are breaking for existing generated clients.  A client that strictly follows the spec will throw errors when new values start coming back for enum fields, and any customer code based on existing releases of this SDK will be broken while they wait for us to publish a new spec and generate/release an SDK update.  If we want customers to depend on our SDKs and associated tools, we need to establish a timeline for changes like these, communicate that to customers, and ensure that we support old & new clients for some period of time.  (It's possible that's already happening for production features and only happening here because there are no customers depending on these endpoints yet; I'm just calling out that this sort of change can't be standard procedure.)

File: fabric/v4/model_cloud_router_action_type.go
Comment (ctreatma) [inline fabric/v4/model_cloud_router_action_type.go:17]: I don't know why the diff rendered this way, but for some reason `git` decided that GatewayActionState became CloudRouterActionType instead of CloudRouterActionState.  That makes it look like there are major enum changes in this release.

Comment (ctreatma) [reply fabric/v4/model_cloud_router_action_type.go:17]: The diff should have compared GatewayActionState to CloudRouterActionState; the old GatewayActionState code is above this comment, and the new CloudRouterActionState code is here: https://github.com/equinix-labs/fabric-go/pull/17/files#diff-564860f61a10f2726340a09cd95a2e0bb507c481e5ad00acf65d1b835f5a3fb6R17-R19

**Review by ctreatma — COMMENTED** [review]
This all looks like it generated correctly, but someone on the Fabric side should weigh in as to whether this correctly implements the expected changes.

File: fabric/v4/model_route_table_entry_filters.go
Comment (thogarty) [reply fabric/v4/model_route_table_entry_filters.go:6]: We've made a comment and will add that to the backlog for the next release. Great comment 👍

**Comment by srushti-patl** [comment]
LGTM!

File: fabric/v4/model_cloud_router_action_type.go
Comment (thogarty) [reply fabric/v4/model_cloud_router_action_type.go:17]: That's something I've noticed about the SDK updates as well. It's a little funky but it does end up in our desired state. We've updated the changes in the description of the PR and are moving forward with it. Thanks for the comments, Charles!

**Review by srushti-patl — APPROVED** [review]

**Pull request merged**
